worlds/civ_6/Rules.py

```diff
@@ -0,0 +1,63 @@
+import typing
+
+from BaseClasses import CollectionState
+from .Items import get_item_by_civ_name
+from .Data import get_boosts_data
+from .Enum import CivVICheckType
+from .ProgressiveDistricts import convert_items_to_have_progression
+
+from worlds.generic.Rules import forbid_item, set_rule
+
+
+if typing.TYPE_CHECKING:
+    from . import CivVIWorld
+
+
+def generate_has_required_items_lambda(prereqs: typing.List[str], required_count: int, has_progressive_items: bool, player: int):
+    def has_required_items_lambda(state: CollectionState):
+        return has_required_items(state, prereqs, required_count, has_progressive_items, player)
+    return has_required_items_lambda
```

Comment:
Ah nice call. I'm unsure if there's a better way to do this. oOriginally when I had this function as just a lambda inside the `set_rule` call, it would not be able to use the value of the `prereq` or `required count` at runtime since those values were overriden by the time it was called. Defining the lambda inside a scoped function in this way allows it to take variables that will always be in context when that lambda is eventually called.

I'm very open to the likelihood there is a less arcane way to do this though 😂

Reply:
*Definitely updating to not pass the unnecessary stuff in 🔥

Reply:
`has_required_items` looks like it should work directly I think?

Reply:
Eh having to pass in the required_count and the prereqs janks it up, I'll take another spin at it though to see if I am just misremembering